src/shogun/mathematics/linalg/LinalgBackendEigen.h

```diff
@@ -127,6 +128,18 @@ class LinalgBackendEigen : public LinalgBackendBase
 		return c;
 	}
 
+	/** Eigen3 matrix C = alpha*A + beta*B method */
+	template <typename T>
+	SGMatrix<T> add_impl(const SGMatrix<T>& a, const SGMatrix<T>& b, T alpha, T beta) const
```

Comment:
so think about
```
float64_t x[2048];
for (size_t i = 0; i < 2048; ++i) {
   x[i] = function(x[i]);
}
```
this is kind of the logic of inplace right? when you do this you do a `read x[i]`, `write x[i]` and then `read [i+1]` so forth and so on... since you do a `write x[i]`, it'll have to go into the RAM that will take:
 a) time
 b) will invalidate the L1 cache
so yeah it's gonna take much more time than:
```
float64_t in[2048];
float64_t out[2048];
for (size_t i = 0; i < 2048; ++i) {
   out[i] = function(in[i]);
}
```